src/components/Button/index.js

```diff
@@ -161,94 +161,127 @@ const defaultProps = {
     forwardedRef: undefined,
 };
 
-class Button extends Component {
-    constructor(props) {
-        super(props);
-
-        this.renderContent = this.renderContent.bind(this);
-    }
-
-    componentDidMount() {
-        if (!this.props.pressOnEnter) {
-            return;
+function Button({
+    allowBubble,
+    text,
+    shouldShowRightIcon,
+
+    icon,
+    iconRight,
+    iconFill,
+    iconStyles,
+    iconRightStyles,
+
+    small,
+    large,
+    medium,
+
+    isLoading,
+    isDisabled,
+
+    onPress,
+    onLongPress,
+    onPressIn,
+    onPressOut,
+    onMouseDown,
+
+    pressOnEnter,
+    enterKeyEventListenerPriority,
+
+    style,
+    innerStyles,
+    textStyles,
+
+    shouldUseDefaultHover,
+    success,
+    danger,
+    children,
+
+    shouldRemoveRightBorderRadius,
+    shouldRemoveLeftBorderRadius,
+    shouldEnableHapticFeedback,
+
+    isFocused,
+    nativeID,
+    accessibilityLabel,
+    forwardedRef,
+}) {
+    useEffect(() => {
+        if (!pressOnEnter) {
+            return () => {};
         }
 
         const shortcutConfig = CONST.KEYBOARD_SHORTCUTS.ENTER;
 
         // Setup and attach keypress handler for pressing the button with Enter key
-        this.unsubscribe = KeyboardShortcut.subscribe(
+        return KeyboardShortcut.subscribe(
             shortcutConfig.shortcutKey,
             (e) => {
-                if (!validateSubmitShortcut(this.props.isFocused, this.props.isDisabled, this.props.isLoading, e)) {
+                if (!validateSubmitShortcut(isFocused, isDisabled, isLoading, e)) {
                     return;
                 }
-                this.props.onPress();
+                onPress();
             },
             shortcutConfig.descriptionKey,
             shortcutConfig.modifiers,
             true,
-            this.props.allowBubble,
-            this.props.enterKeyEventListenerPriority,
+            allowBubble,
+            enterKeyEventListenerPriority,
             false,
         );
-    }
 
-    componentWillUnmount() {
-        // Cleanup event listeners
-        if (!this.unsubscribe) {
-            return;
-        }
-        this.unsubscribe();
-    }
+        // This effect should run only once during mounting
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
 
-    renderContent() {
-        if (this.props.children) {
-            return this.props.children;
+    const renderContent = () => {
```

Comment:
consider memoizing this

Reply:
I'm not sure if it's worth in this case, the dependency list would be huge and very susceptible to re-renders

`React Hook useMemo has missing dependencies: 'children', 'danger', 'icon', 'iconFill', 'iconRight', 'iconRightStyles', 'iconStyles', 'isLoading', 'large', 'medium', 'shouldShowRightIcon', 'small', 'success', 'text', and 'textStyles'. Either include them or remove the dependency array.`

wdyt?